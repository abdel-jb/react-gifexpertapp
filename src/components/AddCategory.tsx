import React from 'react';

type TAddCategory = {
    setCategories: React.Dispatch<React.SetStateAction<string[]>> | any;
}

export const AddCategory = ({ setCategories }: TAddCategory): JSX.Element => {

    const [inputValue, setInputValue] = React.useState('');

    const handleInputChanhe = (e: React.FormEvent<HTMLInputElement>): void => {
        const { value } = e.target as HTMLInputElement;
        setInputValue(value);
    }

    const handleSubmit = (e: React.FormEvent): void => {
        e.preventDefault();
        if (inputValue.trim().length > 2) {
            setCategories(( category: string[] ) => (!category.find(e => e === inputValue) ? [inputValue, ...category] : category));
            setInputValue('');
        }
    };

    return (
        <form action="" onSubmit={ handleSubmit }>
            <input 
                type="text"
                value={ inputValue }
                onChange={ handleInputChanhe }
            />
        </form>
    )
}
