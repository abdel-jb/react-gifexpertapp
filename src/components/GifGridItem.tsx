import React from 'react';

import { TGifs } from '../utils/types';


export const GifGridItem = ( { title, url }: TGifs ): JSX.Element => {
    return (
        <li className="card animate__animated animate__fadeIn">
            <img src={ url } alt={ title }/>
            <p>{ title }</p>
        </li>
    )
}
