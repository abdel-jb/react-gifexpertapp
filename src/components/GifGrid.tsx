import React from 'react';

import { GifGridItem } from './GifGridItem';
import { TGifs } from '../utils/types';
import { useFetchGifs } from '../hooks/useFetchGifs';


type TGifGrid = {
    category: string
}

export const GifGrid = ({ category }: TGifGrid): JSX.Element => {

    const { data: images, loading } = useFetchGifs(category);

    return (
        <>
            <h3>{ category }</h3>

            { loading && <p>Loading...</p> }

            <ul className="card-grid">
                { images.map(( img: TGifs ) => 
                    <GifGridItem
                        key={ img.id }
                        { ...img }
                    />
                )}
            </ul>
        </>
    )
}
