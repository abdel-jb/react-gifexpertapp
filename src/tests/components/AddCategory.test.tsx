import React from 'react';
import { shallow, ShallowWrapper } from "enzyme";
import { AddCategory } from '../../components/AddCategory';

describe('Test in AddCategory', () => {

    const setCategories = jest.fn();
    let wrapper: ShallowWrapper<any, Readonly<{}>, React.Component<{}, {}, any>>;

    beforeEach(() =>{
        jest.clearAllMocks();
        wrapper = shallow(<AddCategory setCategories/>);
    });

    test('should display correctly', () => {

        expect( wrapper ).toMatchSnapshot();

    });

    test('should change input', () => {
        
        const input = wrapper.find('input');
        const value = 'Hello World';

        input.simulate('change', { target: { value } });

        const inputAfter = wrapper.find('input');

        expect( inputAfter.prop('value') ).toBe( value );

    });

    test('should not post the information with submit', () => {

        wrapper.find('form').simulate('submit', { preventDefault(){} });

        expect( setCategories ).not.toHaveBeenCalled();

    });

    // test('should call setCategories and clear input', () => {

    //     const input = wrapper.find('input');
    //     const value = 'Testing';

    //     input.simulate('change', { target: { value } });

    //     wrapper.find('form').simulate('submit', { preventDefault(){} });

    //     expect( setCategories ).toHaveBeenCalled();

    //     expect( setCategories ).toHaveBeenCalledWith( expect.any(Function) );

    //     expect( input.prop('value') ).toBe('');


    // });
    
    
    
});
