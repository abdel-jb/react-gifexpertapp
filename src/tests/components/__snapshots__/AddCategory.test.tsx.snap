// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in AddCategory should display correctly 1`] = `
<form
  action=""
  onSubmit={[Function]}
>
  <input
    onChange={[Function]}
    type="text"
    value=""
  />
</form>
`;
